src/sentry/remote_config/endpoints.py

```diff
@@ -0,0 +1,73 @@
+from rest_framework import serializers
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.serializers import Serializer
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectEndpoint, ProjectEventPermission
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.models.project import Project
+from sentry.models.projectkey import ProjectKey
+from sentry.remote_config.storage import make_storage
+
+
+class ConfigurationValidator(Serializer):
+    id = serializers.UUIDField(read_only=True)
+    sample_rate = serializers.FloatField(max_value=1.0, min_value=0, required=True)
+    traces_sample_rate = serializers.FloatField(max_value=1.0, min_value=0, required=True)
+    user_config = serializers.JSONField(required=True, allow_null=True)
+
+
+class ConfigurationContainerValidator(Serializer):
+    data = ConfigurationValidator()  # type: ignore[assignment]
+
+
+@region_silo_endpoint
+class ProjectConfigurationEndpoint(ProjectEndpoint):
+    owner = ApiOwner.REMOTE_CONFIG
+    permission_classes = (ProjectEventPermission,)
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+        "POST": ApiPublishStatus.EXPERIMENTAL,
+        "DELETE": ApiPublishStatus.EXPERIMENTAL,
+    }
+
+    def convert_args(self, request: Request, key_id: str, *args, **kwargs):
+        args, kwargs = super().convert_args(request, *args, **kwargs)
+        project = kwargs["project"]
+
+        try:
+            key = ProjectKey.objects.for_request(request).get(project=project, public_key=key_id)
+        except ProjectKey.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        kwargs["key"] = key
+        return args, kwargs
+
+    def get(self, request: Request, project: Project, key: ProjectKey) -> Response:
+        """Get remote configuration from project options."""
+        remote_config = make_storage(key).get()
```

Comment:
Is it worth caching the StorageBackends with @lru_cache or something?

Reply:
Nah, we're only initializing a class.  The memory use from caching would be more damaging than the CPU cost of creating a new instance.

Reply:
Cool cool, `get_storage` in the driver is the same thing right, just an import?

Reply:
Why do we have the `make_storage` function instead of directly using the `StorageBackend` constructor?

Reply:
Its indirection solely for the sake of abstracting the storage definition.  I had another storage driver previously and may add another one in the future.  This makes it easier for me to change things internally without worrying about the implementation in the test suite and in the endpoints.

It can be removed once things have stabilized.